src/languages/lua/lua.test.ts

```diff
@@ -8,87 +8,85 @@ import {
   stringLiteral,
   id,
   polygolfOp,
-  print,
+  arrayConstructor,
+  Expr,
 } from "../../IR";
-import { applyLanguage } from "../../common/applyLanguage";
+import applyLanguage from "../../common/applyLanguage";
 
 function expectTransform(program: IR.Program, output: string) {
   expect(applyLanguage(lua, program)).toEqual(output);
 }
 
-function expectStatement(statement: IR.Statement, output: string) {
+function expectStatement(statement: IR.Expr, output: string) {
   expectTransform(program(block([statement])), output);
 }
 
-function testStatement(desc: string, statement: IR.Statement, output: string) {
-  test(desc, () => expectStatement(statement, output));
-}
-
 function testpolygolfOp(
   op: IR.OpCode,
-  args: (IR.Expr | string)[],
+  args: ("i" | "I" | "t" | "T" | "b" | "B" | "a" | Expr)[],
   output: string
 ) {
-  testStatement(
-    op,
-    polygolfOp(op, ...args.map((x) => (typeof x === "string" ? id(x) : x))),
-    output
+  test(op, () =>
+    expectTransform(
+      program(
+        block([
+          assignment("i", int(0n)),
+          assignment("I", int(4n)),
+          assignment("t", stringLiteral("abc")),
+          assignment("T", stringLiteral("DEF")),
+          assignment("b", polygolfOp("true")),
+          assignment("B", polygolfOp("false")),
+          assignment("a", arrayConstructor([stringLiteral("xy")])),
+          polygolfOp(
+            op,
+            ...args.map((x) => (typeof x === "string" ? id(x) : x))
+          ),
+        ])
+      ),
+      `i=0
+I=4
```

Comment:
Smart way to bring the types into the tests without complicating the describe blocks.